File: lab1 stack, queue, deque/B.py
import queue

q = queue.Queue()

while True:
    inp = input().split()

    if inp[0] == 'push':
        q.put(inp[1])
        print('ok')
    elif inp[0] == 'pop':
        if not q.empty():
            print(q.get())
        else: print('error')
    elif inp[0] == 'front':
        if not q.empty():
            print(q.queue[0])
        else: print('error')
    elif inp[0] == 'size':
        print(q.qsize())
    elif inp[0] == 'clear':
        while not q.empty():
            q.get()
        print('ok')
    elif inp[0] == 'exit':
        print('bye')
        exit()

File: lab1 stack, queue, deque/B.cpp
#include <iostream>
#include <queue>

using namespace std;

int main(){
    queue<int> q;
    int n;

    while (true){
        string s;
        cin >> s;

        if (s == "push"){
            cin >> n;
            q.push(n);
            cout << "ok" << endl;
        }

        else if (s == "pop"){
            if (!q.empty()){
                cout << q.front() << endl;
                q.pop();
            }
            else
                cout << "error" << endl;
        }

        else if (s == "front"){
            if (!q.empty()){
                cout << q.front() << endl;
            }
            else
                cout << "error" << endl;
        }

        else if (s == "size"){
            cout << q.size() << endl;
        }

        else if (s == "clear"){
            while (!q.empty())
            q.pop();
            cout << "ok" << endl;
        }

        else if (s == "exit"){
            cout << "bye";
            return 0;
        }
    }


}

